### DISCUSSION TITLE
Auto Dark Mode

### DISCUSSION THREAD

**Discussion by alexjoyner** [tailwindlabs/tailwindcss#6552] [Ideas]
When using dark mode, in my experience most of the dark color theme is just, for the most part, the light color styles inverted.  Is it possible we could have some sort of autoDarkMode: true and it uses either a custom or pre-defined set of mappings for creating an automatically enabled dark mode. That way we would, excluding little edge case things, get a rough version of dark mode instantly and would only have to worry about edge case stuff with the actual dark classes instead of manually adding it to every instance of every color?

For example:
```
               light mode  ==>    dark mode
               bg-gray-50  ==>  bg-gray-900
                 100 ==>                 800
                 200 ==>                 700
                 300 ==>                 600
                 400 ==>                 500
                 500 ==>                 400
                        ...........  <==
```
Just using this simple rule, I applied it to the tailwind playground. Other than a couple of little tweaks, it seems promising to me.
With something like this enabled here, all I would have needed to do is enable dark mode and I could remove all the dark: stuff
![image](https://user-images.githubusercontent.com/7675099/146284964-76034ee2-0a9c-449f-91b3-a5f93884237c.png)

This one is with the home page example but slightly updated with more consistent inverting. In this case, all you would need to do is sprinkle in a couple of overrides to not invert the play button and to make the start time white to be really close to the example.
![image](https://user-images.githubusercontent.com/7675099/146295212-3ecb3735-8807-4353-a0dc-de5bd88cf57a.png)

**Comment by sovetski** [comment]
A big yes to this proposal, I'm surprised there hasn't been any news on this topic since 2021. I know some people will say "yeah, but it's not recommended, etc." but most of the time we just need a quick dark mode without overcomplicating things. For a simple dark mode, we have to add "dark:" and so on, which makes the code even more verbose.
